tests/e2e/volume_provisioning_with_level5_topology.go

```diff
@@ -0,0 +1,1237 @@
+		gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset should match with number of replicas")
+
+		/* Verify PV node affinity and that the PODS are running on appropriate
+		node as specified in the allowed topologies of SC */
+		ginkgo.By("Verify PV node affinity and that the PODS are running on appropriate " +
+			"node as specified in the allowed topologies of SC")
+		verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulset, namespace, allowedTopologies)
+
+		// Scale up statefulset replicas to 5
+		replicas += 5
+		ginkgo.By("Scale up statefulset replica count to 5")
+		scaleUpStatefulSetPod(ctx, client, statefulset, namespace, replicas)
+
+		// Scale down statefulset replicas to 1
+		replicas -= 1
+		ginkgo.By("Scale down statefulset replica count to 1")
+		scaleDownStatefulSetPod(ctx, client, statefulset, namespace, replicas)
+
+		/* Verify newly created PV node affinity and that the new PODS are running on
+		appropriate node as specified in the allowed topologies of SC */
+		ginkgo.By("Verify newly created PV node affinity and that the new PODS are running " +
+			"on appropriate node as specified in the allowed topologies of SC")
+		verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulset, namespace, allowedTopologies)
+
+		// Scale down statefulset replicas to 0
+		replicas = 0
+		ginkgo.By("Scale down statefulset replica count to 0")
+		scaleDownStatefulSetPod(ctx, client, statefulset, namespace, replicas)
+	})
+
+	/*
+	   TESTCASE-7
+	   Steps:
+	   1. Create SC with Immediate BindingMode with all 5 levels of allowedTopologies and storage policy specific to rack3.
+	   (here in this case - region1 > zone1 > building1 > level1 > rack > rack3)
+	   2. Create PVC using above SC.
+	   3. Create Deployment set with replica count 1 using above created PVC.
+	   4. Vefiry Deployment pod is in up and running state and PVC in bound phase.
+	   5. Describe PV and verify pv node affinity should have allowed topology details of SC.
+	   6. Verify pod is created on the node as mentioned in the storage class.
+	   7. Delete Deployment set
+	   8. Delete PVC and SC
+	*/
+	ginkgo.It("Provisioning volume when storage class specified with "+
+		"Immediate BindingMode with multiple allowed topologies and using DeploymentSet pod", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var lables = make(map[string]string)
+		lables["app"] = "nginx"
+		replica := 1
+
+		/* Get allowed topologies for Storage Class
+		(region1 > zone1 > building1 > level1 > rack > rack3) */
+		allowedTopologyForSC := getTopologySelector(topologyAffinityDetails, topologyCategories,
+			topologyLength, leafNode, leafNodeTag2)
+
+		// create SC and PVC
+		storagePolicyName := GetAndExpectStringEnvVar(envStoragePolicyNameForNonSharedDatastores)
```

Comment:
How can volume provision can happen in non-shared datastore

Reply:
Volume provisioning will happen in this case on cluster2 (rack2) because I have considered ds specific to  cluster2 and I am using already created env. parameter - envStoragePolicyNameForNonSharedDatastores.

If not this name, Could you please provide new env. paramter name which can provision volume on any 1 specific cluster.